src/pending_tasks/api/views.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from django.db.models import Q
+from djgentelella.objectmanagement import AuthAllPermBaseObjectManagement
+from rest_framework.decorators import action
+from rest_framework.permissions import BasePermission
+from rest_framework.response import Response
+from rest_framework import status
+
+from pending_tasks.models import PendingTask
+from pending_tasks.api.serializers import PendingTaskSerializer, \
+    PendingTaskListSerializer, PendingTaskValidateSerializer
+
+logger = logging.getLogger("django")
+
+
+class PendingTaskViewSet(AuthAllPermBaseObjectManagement):
+    queryset = PendingTask.objects.all()
+
+    serializer_class = {
+        'list': PendingTaskListSerializer,
+        'destroy': PendingTaskSerializer,
+        'create': PendingTaskValidateSerializer,
+        'update': PendingTaskValidateSerializer,
+    }
+
+    permission_classes = (BasePermission,)
+
+    ordering = ('creation_date',)
+
+    def get_queryset(self):
+        return PendingTask.objects.filter(
+            Q(validate_profile=self.request.user.profile) | Q(
+                validate_profile__isnull=True))
+
+    @action(detail=True)
+    def task_assign(self, request, *args, **kwargs):
+        instance = self.get_object()
+        if instance.validate_profile is None:
+            instance.validate_profile = request.user.profile
```

Comment:
You need to create a validate serializer profile, profile field is required and you need to verify that profile belong to an organization form PendingTask model.

The logger warning could you move inside the serializer validation.
The logger info is not necessary, instead you need to import the organilab_logentry util function.